installation/downloadReferences.sh

```diff
@@ -1,66 +1,105 @@
+#!/bin/bash
 
+set -o pipefail
+
+# which files to download
 REFERENCE_GENOME=true
 dbSNP_FILE=true
 MAPPABILITY_FILE=true
 CHROMOSOME_LENGTH_FILE=true
 statFiles=true
-IMPUTE_FILES=false
-
-SCRIPT=$(readlink -f $0)
-scriptdir=`dirname $SCRIPT`
+IMPUTE_FILES=true
 
-echo $scriptdir
+exit_on_fail() {
+        echo "Download incomplete. Please restart script." 1>&2
+        exit 1
+}
 
-mkdir -p sequence/1KGRef
-mkdir databases
-mkdir stats
-mkdir databases/ENCODE
-mkdir -p databases/1000genomes/IMPUTE
-mkdir databases/UCSC
-mkdir -p databases/dbSNP/dbSNP_135
+mkdir -p hg19_GRCh37_1000genomes &&
+cd hg19_GRCh37_1000genomes ||
+exit_on_fail
 
-if [[ $REFERENCE_GENOME != "true" ]] 
+if [[ "$REFERENCE_GENOME" == "true" && ! -e sequence/1KGRef/hs37d5.fa ]] 
 then
-	echo downloading reference genome///
-        wget -P sequence/1KGRef ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.fa.gz
-	gunzip sequence/1KGRef/hs37d5.fa.gz
+	echo downloading reference genome....
+	mkdir -p sequence/1KGRef &&
+	wget -c -P sequence/1KGRef ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.fa.gz &&
+	gunzip sequence/1KGRef/hs37d5.fa.gz ||
+	exit_on_fail
 fi
 
-if [[  $dbSNP_FILE != "true" ]]
+if [[ "$dbSNP_FILE" == "true" && ! -e databases/dbSNP/dbSNP_135/00-All.SNV.vcf.gz.tbi ]]
 then
-	echo dbSNP file....
-       installationpath=`pwd`
-       cd databases/dbSNP/dbSNP_135
-       bash $scriptdir/prepare_dbSNPFile.sh
-	cd $installationpath
+	echo downloading dbSNP file....
+	mkdir -p databases/dbSNP/dbSNP_135 &&
+	cd databases/dbSNP/dbSNP_135 &&
```

Comment:
Probably better that `cd $path` and `cd -` are `pushd $path` and `popd`. At least they are more expressive to the reader.

Reply:
You can also get rid of cd -, if you wrap the whole block into its own process. You only need to make sure, that all variables are exported beforehand. And of course, you can't fill outside variables inside the sub process.